### ISSUE TITLE
Docs show config options that appear to have been removed

### ISSUE THREAD

**Issue by arizonatribe** [styleguidist/react-styleguidist#1292]
**Current behavior**

While working with the Node API for React Styleguidist, I noticed a couple of times where I was receiving the "Something is wrong with your style guide config", but no other info. Tracing that [in the code](https://github.com/styleguidist/react-styleguidist/blob/master/src/scripts/config.js#L49) it looks like there's just situations where there is no `exception.message` and `exception.extra` to accompany the message. Locally I added a `console.log` to that file in my `node_modules/react-styleguidist/` location to see if there's anything useful on the `exception` itself, and it gave the full information on the field name that it didn't like.

```
Unknown config option moduleAliases was found, the value is: // rest of output
```

To be clear, I'm not opening the issue for that unsupported field. It looks like there was a hiccup in the support for that field: seems to have been removed from the `schema/config.js` in the version 8.0.6 I was using, but I see it's back now in 9.0.1 (one oddity still is [the docs still have it listed on the page](https://react-styleguidist.js.org/docs/configuration.html#modulealiases), just not among the table-of-contents on [the top of the page](https://react-styleguidist.js.org/docs/configuration.html)).

The reason for the issue is the logging of the error seems like it could be improved (or fixed, depending on your opinion):

```js
if (exception instanceof StyleguidistError) {
  throw new StyleguidistError(
    'Something is wrong with your style guide config',
    exception.message,
    exception.extra
  );
}
```
That's the current code (in the `scripts/config.js`) but `exception.message` doesn't seem to exist on [the `StyleguidistError` object](https://github.com/styleguidist/react-styleguidist/blob/master/src/scripts/utils/error.js). The `extra` prop _is_ part of that custom error, but the "message" is just used to feed into the underlying (native) `Error`.

There's really nothing to catch the error and pull the `.extra` prop off of it, unless the user _knows_ to do so. It seems like it makes more sense that if you have the reason for the error, to show that instead as the message.

```js
if (exception instanceof StyleguidistError) {
  throw new StyleguidistError(
    exception.message || 'Something is wrong with your style guide config',
    exception.extra
  );
}
```

Thanks!

**Comment by sapegin** [comment]
The `moduleAliases` was introduced in 9.x, so we need to update the TOC in the docs ;-/

Does it work fine with `exception.toString()`? I'm fine with any solution if it improves the experience.

**Comment by arizonatribe** [comment]
Actually one minor change to that recommendation. I noticed the constructor for the `StyleguidistError` expects `(message, extra, anchor) -> *`, but the way it's being used in the `scripts/config.js` might be expecting `(fallback, message, extra) -> *`?

Editing the `node_modules/react-styleguidist/lib/scripts/config.js` locally like this seemed to work best:
```js
if (exception instanceof StyleguidistError) {
  throw new StyleguidistError(
    exception.toString() || 'Something is wrong with your style guide config',
    exception.extra
  );
}
```

**Comment by sapegin** [comment]
Sounds good! Mind sending a PR with this change?

**Comment by arizonatribe** [comment]
Yeah, but I can't seem to find 9.0.1. When I fork and pull the latest `master` it's still at 9.0.0

**Comment by arizonatribe** [comment]
I can go ahead and bump to 9.0.2 and submit that as the PR

**Comment by sapegin** [comment]
Ah, that's because of semantic-release. It doesn't change the version number in `package.json` (it should be `0.0.0-development`) and I've published 9.0.0 manually. So you don't need to care about versions.

**Comment by arizonatribe** [comment]
Okay great; submitted the PR for those two
